crates/core/src/operations/update.rs

```diff
@@ -568,19 +594,47 @@ pub(crate) async fn request_update(
             .pop()
             .ok_or(OpError::RingError(RingError::NoLocation))?
     } else {
+        // Check if we have any other peers that can cache contracts
         let closest = op_manager
             .ring
             .closest_potentially_caching(key, [sender.peer.clone()].as_slice())
             .into_iter()
-            .next()
-            .ok_or_else(|| RingError::EmptyRing)?;
+            .next();
 
-        op_manager
-            .ring
-            .add_subscriber(key, sender)
-            .map_err(|_| RingError::NoCachingPeers(*key))?;
+        if closest.is_none() {
+            // Check if we actually have any connected peers at all
+            let has_connections = op_manager.ring.connection_manager.num_connections() > 0;
```

Comment:
again, if you are not connected to anybody, why would you be able to do updates at all?